src/solidity-frontend/solidity_convert.cpp

```diff
@@ -1506,6 +1506,20 @@ bool solidity_convertert::get_noncontract_defition(nlohmann::json &ast_node)
     // for abstract contract
     add_empty_body_node(ast_node);
   }
+  else if (node_type == "ContractDefinition" && ast_node["contractKind"] == "library")
+  {
+    if (get_struct_class(ast_node))
+      return true;
+    if (convert_ast_nodes(ast_node))
+      return true;
+    // static instance
+    std::string lib_name = ast_node["name"].get<std::string>();
+    symbolt lib_instance;
+    get_static_contract_instance(lib_name, lib_instance);
+    lib_instance.static_lifetime = true;
```

Comment:
You don't need to specify it's `static`. they are handled in the `get_static_contract_instance`.

Reply:
I feel like you need to improve the get_function_params,  to support `storage`.